src/cascadia/TerminalSettingsModel/Command.h

```diff
@@ -102,15 +103,13 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         WINRT_PROPERTY(OriginTag, Origin);
         WINRT_PROPERTY(winrt::hstring, Description, L"");
 
-    public:
-        til::typed_event<Model::Command, winrt::hstring> IDChanged;
-
     private:
         Json::Value _originalJson;
         Windows::Foundation::Collections::IMap<winrt::hstring, Model::Command> _subcommands{ nullptr };
         std::optional<CommandNameOrResource> _name;
         std::wstring _ID;
         bool _IDWasGenerated{ false };
+        std::function<void(const Model::Command, const std::wstring_view)> _pfnIDChanged;
```

Comment:
is it possible to not use a callback-based model at all? just have the top level settings object handle rename and propagating every necessary ID change downward? it would make command stateless and reduce back-channeling between all our objects :/

Reply:
updated!